StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1503UnitTests.cs

```diff
@@ -0,0 +1,466 @@
+    {
+        if (i == 0) if (i == 0) Debug.Assert(true);
+    }
+}";
+
+            var fixedTestCode = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        if (i == 0)
+        {
+            if (i == 0)
+            {
+                Debug.Assert(true);
+            }
+        }
+    }
+}";
+
+            await this.VerifyCSharpFixAsync(testCode, fixedTestCode);
+        }
+
+        private async Task TestStatementWithoutCurlyBrackets(string statementText)
+        {
+            var expected = new[]
+            {
+                new DiagnosticResult
+                {
+                    Id = DiagnosticId,
+                    Message = "Curly brackets must not be omitted",
+                    Severity = DiagnosticSeverity.Warning,
+                    Locations = new[] { new DiagnosticResultLocation("Test0.cs", 7, 13) }
+                }
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(this.GenerateTestStatement(statementText), expected, CancellationToken.None);
+        }
+
+        private async Task TestStatementWithCurlyBrackets(string statementText)
+        {
+            await this.VerifyCSharpDiagnosticAsync(this.GenerateFixedTestStatement(statementText), EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        private async Task TestCodeFixForStatement(string statementText)
+        {
+            await this.VerifyCSharpFixAsync(this.GenerateTestStatement(statementText), this.GenerateFixedTestStatement(statementText));
+        }
+
+        private string GenerateTestStatement(string statementText)
+        {
+            var testCodeFormat = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        #STATEMENT#
+            Debug.Assert(true);
+    }
+}";
+            return testCodeFormat.Replace("#STATEMENT#", statementText);
+        }
+
+        private string GenerateFixedTestStatement(string statementText)
+        {
+            var fixedTestCodeFormat = @"using System.Diagnostics;
+public class Foo
+{
+    public void Bar(int i)
+    {
+        #STATEMENT#
+        {
+            Debug.Assert(true);
+        }
+    }
+}";
+            return fixedTestCodeFormat.Replace("#STATEMENT#", statementText);
+        }
+
+        /// <inheritdoc/>
+        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+        {
+            return new SA1503CurlyBracketsMustNotBeOmitted();
+        }
+
+        /// <inheritdoc/>
+        protected override CodeFixProvider GetCSharpCodeFixProvider()
+        {
+            return new SA1503CodeFixProvider();
+        }
+
+        protected override Solution CreateSolution(ProjectId projectId, string language)
+        {
+            var newSolution = base.CreateSolution(projectId, language);
+
+            newSolution.Workspace.Options = newSolution.Workspace.Options.WithChangedOption(FormattingOptions.IndentationSize, language, this.indentSize);
```

Comment:
See pull request #522.
I'm leaving the code in here for now so that this can still be built.